src/rttest.cpp

```diff
@@ -696,21 +766,48 @@ int rttest_calculate_statistics(struct rttest_results * results)
   return thread_rttest_instance->calculate_statistics(results);
 }
 
-int rttest_get_statistics(struct rttest_results & output)
+int rttest_get_statistics(struct rttest_results * output)
 {
   auto thread_rttest_instance = get_rttest_thread_instance(pthread_self());
   if (!thread_rttest_instance) {
     return -1;
   }
-  output = thread_rttest_instance->results;
+  if (output == NULL) {
+    output = &thread_rttest_instance->results;
+  } else {
+    // if output is not null, try to copy the results struct into the memory location
+    *output = thread_rttest_instance->results;
+  }
+
   return 0;
 }
 
-std::string rttest_results_to_string(struct rttest_results * results, char * name)
+int Rttest::get_sample_at(const size_t iteration, int & sample) const
 {
-  if (!results) {
-    return "ERROR: rttest got NULL results string!";
+  if (this->params.iterations == 0) {
+    sample = this->sample_buffer.latency_samples[0];
+  } else if (iteration < this->params.iterations) {
+    sample = this->sample_buffer.latency_samples[iteration];
+  } else {
+    return -1;
   }
+  return 0;
```

Comment:
Actually, the first two if statements don't return, so they will go to this return statement. I could change it to be more readable, however.